integration/query_compat_test.go

```diff
@@ -192,6 +196,130 @@ func testQueryCompat(t *testing.T, testCases map[string]queryCompatTestCase) {
 	testQueryCompatWithProviders(t, shareddata.AllProviders(), testCases)
 }
 
+func TestQueryCappedCollectionCompat(t *testing.T) {
+	t.Parallel()
+
+	s := setup.SetupCompatWithOpts(t, &setup.SetupCompatOpts{
+		Providers:                []shareddata.Provider{},
+		AddNonExistentCollection: true,
+	})
+	ctx, targetDB, compatDB := s.Ctx, s.TargetCollections[0].Database(), s.CompatCollections[0].Database()
+
+	cName := testutil.CollectionName(t)
+	opts := options.CreateCollection().SetCapped(true).SetSizeInBytes(1000)
+
+	targetErr := targetDB.CreateCollection(s.Ctx, cName, opts)
+	require.NoError(t, targetErr)
+
+	compatErr := compatDB.CreateCollection(s.Ctx, cName, opts)
+	require.NoError(t, compatErr)
+
+	targetCollection := targetDB.Collection(cName)
+	compatCollection := compatDB.Collection(cName)
+
+	// documents inserted are sorted to ensure the insertion order of capped collection
+	docs := shareddata.Doubles.Docs()
+	slices.SortFunc(docs, func(a, b bson.D) int {
+		aID := must.NotFail(ConvertDocument(t, a).Get("_id")).(string)
+		bID := must.NotFail(ConvertDocument(t, b).Get("_id")).(string)
+		return cmp.Compare(aID, bID)
+	})
+
+	insert := make([]any, len(docs))
+	for i, doc := range docs {
+		insert[i] = doc
+	}
+
+	targetInsertRes, targetErr := targetCollection.InsertMany(ctx, insert)
+	require.NoError(t, targetErr)
+
+	compatInsertRes, compatErr := compatCollection.InsertMany(ctx, insert)
+	require.NoError(t, compatErr)
+
+	require.Equal(t, compatInsertRes, targetInsertRes)
+
+	for name, tc := range map[string]struct {
+		filter bson.D
+		sort   bson.D
+
+		sortPushdown resultPushdown
+	}{
+		"NoSortNoFilter": {
+			sortPushdown: allPushdown,
+		},
+		"Filter": {
+			filter:       bson.D{{"v", int32(42)}},
+			sortPushdown: allPushdown,
+		},
+		"Sort": {
+			sort:         bson.D{{"_id", int32(-1)}},
+			sortPushdown: allPushdown,
+		},
+		"FilterSort": {
+			filter:       bson.D{{"v", int32(42)}},
+			sort:         bson.D{{"_id", int32(-1)}},
+			sortPushdown: allPushdown,
+		},
```

Comment:
As we do some manipulations with `order` pushdown clause for capped collections, I suggest to add one more test case where we have a bit more complicated case, e.g. `sort:  bson.D{{"v", 1}, {"_id", int32(-1)}}` to ensure such cases work correctly too.

Reply:
Added an additional test case.